tools/network.py

```diff
@@ -83,7 +154,39 @@ def fetch_raw(
     request_blocking = QgsBlockingNetworkRequest()
     if authcfg_id:
         request_blocking.setAuthCfg(authcfg_id)
-    _ = request_blocking.get(req)
+    # QNetworkRequest *only* supports get and post. No idea why.
+    if method == "get":
+        _ = request_blocking.get(req)
+    elif method == "post":
+        if data:
+            # Support JSON
+            byte_data = bytes(json.dumps(data), encoding)
+            req.setRawHeader(b"Content-Type", bytes("application/json", encoding))
+        elif files:
+            # Support multipart binary. Generate boundary like
+            # https://github.com/requests/toolbelt/blob/master/requests_toolbelt/multipart/encoder.py
+            boundary = uuid4().hex
+            byte_boundary = bytes(f"\r\n--{boundary}\r\n", encoding)
+            last_byte_boundary = bytes(f"\r\n--{boundary}--\r\n", encoding)
+            byte_boundary_with_disposition = (
+                byte_boundary
+                + bytes(CONTENT_DISPOSITION_FORM_DATA, encoding)
+                + bytes(CONTENT_TYPE_FORM_DATA, encoding)
+            )
+            byte_data = (
```

Comment:
Can you please revisit this code part?
I think now there is an extra `byte_boundary_with_disposition` at the start. Should just be `byte_boundary`
And all the Content-Disposition: form-data input names, filenames and content types are now static in the `byte_boundary_with_disposition` regardless of the file.

Reply:
Ah, I got confused how the join method works. The first point of mine is not relevant. Sorry for that.